src/test/features/response/routes/your-details.ts

```diff
@@ -56,6 +76,66 @@ describe('Defendant user details: your name page', () => {
             .set('Cookie', `${cookieName}=ABC`)
             .expect(res => expect(res).to.be.successful.withText('Confirm your details'))
         })
+
+        it('should render page without firstName when claim doesn\'t have firstName', async () => {
+          draftStoreServiceMock.resolveFind('response')
+          claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+
+          await request(app)
+            .get(pagePath)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withoutText('First Name'))
+        })
+
+        it('should render page without lastName when claim doesn\'t have lastName', async () => {
+          draftStoreServiceMock.resolveFind('response')
+          claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+
+          await request(app)
+            .get(pagePath)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withoutText('Last Name'))
+        })
+
+        it('should render page without firstName when claim doesn\'t have title', async () => {
```

Comment:
please don't use escape characters. use "does not" instead

Reply:
done